debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -103,6 +104,11 @@ public SqlServerStreamingChangeEventSource(SqlServerConnectorConfig connectorCon
 
     @Override
     public void execute(ChangeEventSourceContext context) throws InterruptedException {
+        if (connectorConfig.getSnapshotMode().equals(SnapshotMode.INITIAL_ONLY)) {
```

Comment:
Ideally we'd handle this in the coordinator for all the connectors, but given the slight differences around `Snapshotter` et al., that won't be be easily doable atm. I'll add this as a comment to the general snapshot refactoring issue.